src/CommitmentsV2/SFA.DAS.CommitmentsV2.UnitTests/Application/Commands/AddCohortCommandHandlerTests.cs

```diff
@@ -134,11 +159,18 @@ public AddCohortCommandHandlerTestFixture WithProvider(long providerId)
             return this;
         }
 
-        public async Task<AddCohortResponse> Handle(long accountLegalEntity, long providerId)
+        public async Task<AddCohortResponse> Handle(long accountLegalEntity, long providerId, string courseCode)
         {
             Db.SaveChanges();
-            var command = new AddCohortCommand { AccountLegalEntityId = accountLegalEntity, ProviderId = providerId, ReservationId = Guid.NewGuid() };
```

Comment:
Just curious why have Db.SaveChanges() at the start of the fixture function and the Db.SaveChangesAsync() at the end. Do we not just need it at the end?

Reply:
We need both. We only get the rows written to the database when savechanges is called; until then all changes live in the dbcontext. The first savechanges is to save the setup. If we didn't have this the set up would be in the dbcontext but if the code under test tried to find any of the set up data in the db it wouldn't find it.

Reply:
It does look a a little odd, shouldn't the setup code be responsible for saving it's changes to the DB

Reply:
It could do that, following this pattern:
Add Provider - Save
Add Legal Entity - Save
Add … - Save

but instead the set up Adds what it likes and the code running the tests does the save immediately before running the test. This is safer (as it ensures that all saves are committed prior to running the test without requiring each add method perform a save) and takes less code.